### PR TITLE
Support "Did you mean" mistakes suggestion

### LINKED ISSUE #55: "Did you mean" suggestion when misspelled text
UX would be much better and more comparable with web Google Translate interface, when you app would be able to provide "Did you mean" suggestions when text has wrong, misspelled words. I don't know whether API provides this information but IMHO without this kind of suggestion for end users the app has little sense for me.

Zrzut ekranu z 2020-10-17 15-21-14

### PR DESCRIPTION
**Pull request by rafaelmardojai**
Fix #55 

Just discovered that **py-googletrans** returns this data in `extra_data['possible-mistakes']`.

### Demo
Peek 25-10-2020 12-46

### CODE REVIEW

File: src/window.py [hunk 1]
```diff
@@ -565,6 +572,12 @@ def update_trans_button(self, button, keyboard):
 
         return Gdk.EVENT_PROPAGATE
 
+    def on_mistakes_clicked(self, _button):
+        self.mistakes.set_reveal_child(False)
+        self.src_buffer.set_text(self.trans_mistakes[1])
+        # Run trasnlation again
+        self.translation(None)
```
Comment (rafaelmardojai) [inline src/window.py:579]: It's fine to run the translation again? Since we are already showing the translation without the mistake.

Comment (fushinari) [reply src/window.py:579]: Yes, running the translation again is still a good idea.

**Comment by fushinari** [comment]
Few questions and/or suggestions:
- Wouldn't it have been easier to simply use a GtkInfoBar? This seems like a job specifically suited to it too.
- Shouldn't only the possible mistakes be the buttons? i.e. the "Did you mean: " should just be a label, not a part of the button.
- The "?" at the end should probably be avoided. That's the Google Translate default behavior and avoids unnecessary complications from the src text also ending in "?".

Otherwise, looks amazing! I didn't even know there was an `extra_data` variable. Looks interesting. It gives a lot more goodies too, like "See also", "Synonyms", "Definitions", etc. Nice!

**Comment by rafaelmardojai** [comment]
> Wouldn't it have been easier to simply use a GtkInfoBar? This seems like a job specifically suited to it too.

Yeah, but it always force the buttons to the end.

Captura de pantalla de 2020-10-25 15-06-09

Is that fine?

**Comment by rafaelmardojai** [comment]
This is how looks your suggestions without using GtkInfoBar:
Captura de pantalla de 2020-10-25 15-16-28

**Comment by rafaelmardojai** [comment]
> Yeah, but it always force the buttons to the end.

Now looking it more, probably that way is more balanced?

Captura de pantalla de 2020-10-25 15-32-10

**Comment by fushinari** [comment]
Wouldn't it be better to follow Google Translate behavior and just make it a link? Would look a lot more consistent imo. I feel like buttons aren't suited for the job.

**Comment by rafaelmardojai** [comment]
> Wouldn't it be better to follow Google Translate behavior and just make it a link? Would look a lot more consistent imo. I feel like buttons aren't suited for the job.

Yeah, I wanted that too. But I was lazy to implement the custom CSS, since `GtkLinkButton` is supposed to open uris.

**Comment by rafaelmardojai** [comment]
Also that was the reason to use a flat button in the beginning, but looks like it not work for buttons inside a `GtkButtonBox` (that is what uses `GtkInfoBar`).

**Comment by rafaelmardojai** [comment]
@fushinari Same here. Here we only need to finish the appearance. 
If you have time, feel free to push.

**Comment by fushinari** [comment]
> Yeah, I wanted that too. But I was lazy to implement the custom CSS, since `GtkLinkButton` is supposed to open uris.

Luckily, we don't need to. :)

Screenshot_20201026_111744

Looks like it works. Might need some more testing. py-googletrans or the site itself is doing the escaping for us, so we don't even need that.

**Comment by fushinari** [comment]
Oh yeah, also the above image is an example for why we definitely need to keep the re-translation. When only "dit" is given, the translation done is actually for the word "ditch" even though the suggestion is "did".

**Comment by fushinari** [comment]
Screenshot_20201026_114915

Should I add that icon there? And if I do, which icon is more suitable `error-correct-symbolic` as seen in image or `tools-check-spelling-symbolic` which looks like this: 
image

**Comment by rafaelmardojai** [comment]
I thinks `error-correct-symbolic` is the correct semantically. But `tools-check-spelling-symbolic` looks cooler 😂

**Review by fushinari — APPROVED** [review]
Exactly my feelings. :rofl:  Well, let's leave it at the semantically correct thing then. 

Have you tested it? For me, it seems to work fine now. If all else looks okay, go ahead and merge.

**Comment by TomaszGasior** [comment]
Great work! :)

**Pull request merged**